src/main/scala/org/apache/spark/sql/delta/storage/LogStore.scala

```diff
@@ -139,27 +139,60 @@ object LogStore extends LogStoreProvider
   with Logging {
 
   def apply(sc: SparkContext): LogStore = {
-    apply(sc.getConf, sc.hadoopConfiguration)
+    createLogStore(None, sc.getConf, sc.hadoopConfiguration)
   }
 
   def apply(sparkConf: SparkConf, hadoopConf: Configuration): LogStore = {
-    createLogStore(sparkConf, hadoopConf)
+    createLogStore(None, sparkConf, hadoopConf)
+  }
+
+  def apply(logPath: Path, sc: SparkContext): LogStore = {
+    apply(logPath, sc.getConf, sc.hadoopConfiguration)
+  }
+
+  def apply(logPath: Path, sparkConf: SparkConf, hadoopConf: Configuration): LogStore = {
+    createLogStore(Some(logPath), sparkConf, hadoopConf)
   }
 }
 
 trait LogStoreProvider {
   val logStoreClassConfKey: String = "spark.delta.logStore.class"
-  val defaultLogStoreClass: String = classOf[HDFSLogStore].getName
+  val defaultLogStoreClass: Class[HDFSLogStore] = classOf[HDFSLogStore]
 
   def createLogStore(spark: SparkSession): LogStore = {
     val sc = spark.sparkContext
-    createLogStore(sc.getConf, sc.hadoopConfiguration)
+    createLogStore(None, sc.getConf, sc.hadoopConfiguration)
+  }
+
+  def createLogStore(logPath: Path, spark: SparkSession): LogStore = {
+    val sc = spark.sparkContext
+    createLogStore(Some(logPath), sc.getConf, sc.hadoopConfiguration)
+  }
+
+  def createLogStore(logPath: Option[Path], sparkConf: SparkConf,
```

Comment:
val shortDefaultLogStoreClass = Map(
      "hdfs" -> classOf[org.apache.spark.sql.delta.storage.HDFSLogStore].getName,
      "adls" -> classOf[org.apache.spark.sql.delta.storage.AzureLogStore].getName,
      "s3" -> classOf["org.apache.spark.sql.delta.storage.S3SingleDriverLogStore"].getName)

val fs = path.getFileSystem(hadoopConf)

val logstoreClassKey = "spark.delta.logStore." + fs.getScheme + ".impl"

val logstoreClass = sparkConf.get(logstoreClassKey)

if (logstoreClass == null) {
	logstoreClass = shortDefaultLogStoreClass.get(fs.getScheme)
}

user set conf to spark.delta.logStore.$scheme.impl to tell delta which logstore class will be used for current scheme,  and it is no need to modify delta code if we have a new FileSystem scheme

similar to https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L3420

@zsxwing